src/cascadia/TerminalApp/CascadiaSettingsSerialization.cpp

```diff
@@ -139,11 +149,33 @@ JsonObject CascadiaSettings::ToJson() const
 // - json: an object which should be a serialization of a CascadiaSettings object.
 // Return Value:
 // - a new CascadiaSettings instance created from the values in `json`
-std::unique_ptr<CascadiaSettings> CascadiaSettings::FromJson(JsonObject json)
+std::unique_ptr<CascadiaSettings> CascadiaSettings::FromJson(const Json::Value& json)
 {
     std::unique_ptr<CascadiaSettings> resultPtr = std::make_unique<CascadiaSettings>();
 
-    resultPtr->_globals = GlobalAppSettings::FromJson(json);
+    if (auto globals{ json[GLOBALS_KEY.data()] })
+    {
+        if (globals.isObject())
+        {
+            resultPtr->_globals = GlobalAppSettings::FromJson(globals);
+        }
+    }
+    else
+    {
+        // If there's no globals key in the root object, then try looking at the
+        // root object for those properties instead, to gracefully upgrade.
+        // This will attempt to do the legacy keybindings loading too
+        resultPtr->_globals = GlobalAppSettings::FromJson(json);
+
+        // If we didn't find keybindings in the legacy path, then they probably
+        // don't exist in the file. Create the default keybindings if we
+        // couldn't find any keybindings.
+        auto keybindings{ json[KEYBINDINGS_KEY.data()] };

```

Comment:
could this be `.at()`?

Reply:
nah, looks like Json::Value doesn't define that